wavelink/ext/spotify/__init__.py

```diff
@@ -446,30 +448,14 @@ async def _search(self,
             return SpotifyTrack(data)
 
         elif data['type'] == 'album':
-            album_data: dict[str, Any]= {
-                                        'album_type': data['album_type'],
-                                        'artists': data['artists'],
-                                        'available_markets': data['available_markets'],
-                                        'external_urls': data['external_urls'],
-                                        'href': data['href'],
-                                        'id': data['id'],
-                                        'images': data['images'],
-                                        'name': data['name'],
-                                        'release_date': data['release_date'],
-                                        'release_date_precision': data['release_date_precision'],
-                                        'total_tracks': data['total_tracks'],
-                                        'type': data['type'],
-                                        'uri': data['uri'],
-                                        }
             tracks = []
             for track in data['tracks']['items']:
-                track['album'] = album_data
-                if iterator:
-                    tracks.append(track)
-                else:
-                    tracks.append(SpotifyTrack(track))
-
-            return tracks
+                async with self.session.get(track['href'], headers=self.bearer_headers) as resp:
```

Comment:
I am actually not sure about the reasoning behind this.
Could you please explain it better?

From what I understand the current implementation was able to get album data and the corresponding tracks with only one request

Reply:
My original intent doing this was to get the ISRC of album tracks, that are only provided as full TrackObjects, in order for the correct track to be found. However, now that we know that the searching for "None" is the root issue, I do not believe it is necessary anymore.